projects/2020-q1-preservica-verification.md

```diff
@@ -0,0 +1,78 @@
+# Preservica verification
+
+## Problem
+
+Preservica contains 41,990,996 files, many of which would be expensive or impossible to replace.
+Before we decommission Preservica, we want to check that every file:
+
+*   Has been successfully copied to the new storage service, or;
+*   Is a file we are happy to delete
+
+This document describes the verification process.
+
+
+
+## Concepts
+
+Within Preservica, files are organised into *deliverable units (DUs)*.
+A DU is analogous to a folder in a filesystem hierarchy: DUs can be nested, and each file belongs to exactly one DU.
+
+Each DU has an associated catalogue reference: for example, `b11633682` or `SATIH/43`.
+We also know when each DU was created.
+
+We can get the following metadata about each file from the Preservica SQL database:
+
+*   The UUID that identifies the file within Preservica
+*   The size of the file in bytes
+*   The original filename of the file
+*   The SHA-1 checksum of the file
+*   The deliverable unit that this file belongs to
+
+
+
+## The verification process: a how-to
+
+*   **Look for files that were copied byte-for-byte to the storage service.**
+
+    The Preservica metadata lets us work out where a file would have been copied to in the storage service, if it was copied -- space, external identifier, and filename within the bag.
+
+    If that bag exists, and it contains a file with matching filename, file size, and SHA-1 checksum, this file has been copied successfully.
```

Comment:
How do you know that the file within the bag has copied successfully and has not corrupted? Doesn't this only test that the filenames and sizes match, rather than using the SHA-1 hash to verify the continued integrity of the transferred file?

Reply:
Possibly this sentence isn't clear enough? How about:

> If that bag exists, look inside the bag for a file with a matching filename. If such a file exists, and the size and SHA-1 checksum of the file in the storage service match those of the file in Preservica, then this file has been copied successfully.

Reply:
Better. What wasn't clear to me was the checksum supplied by Preservica is verified against a checksum generated using the same algorithm as part of the bagging process (I think that's what you're saying?). I read it as the checksum supplied by Preservica had been copied and pasted - so to speak - into the bag.